ironfish/src/consensus/verifier.ts

```diff
@@ -187,16 +187,7 @@ export class Verifier {
    * the mempool and rebroadcasted to the network.
    */
   async verifyNewTransaction(transaction: Transaction): Promise<VerificationResult> {
-    try {
-      // Transaction is lazily deserialized, so we use takeReference()
-      // to force deserialization errors here
-      transaction.takeReference()
-      transaction.returnReference()
-    } catch (e) {
-      return { valid: false, reason: VerificationResultReason.DESERIALIZATION }
-    }
-
-    // Next, check the spends are valid and not already in the chain
+    // Check the spends are valid and not already in the chain
```

Comment:
This will let crashes through potentially to through the rest of the code base. If you think this is acceptable, then I'm fine merging it but I feel like removing validation doesn't feel really safe.

Reply:
That's incorrect -- the same verification is being performed, but removing a duplicate step (constructing a TransactionPosted on the main thread, then constructing a TransactionPosted in the worker pool).

However, this change might allow invalid data to go to spend verification. I can't imagine a scenario where the original transaction would be parsed without crashing in the `Transaction` constructor, then end up crashing in spend verification -- spend verification is mostly just comparing the equality of buffers. But regardless, I moved the worker pool verification before it.